### PR TITLE
Build blackbox_exporter with go 1.19.9 

### PR DESCRIPTION
**Pull request by electron0zero**
Go came out with new secruity release with [fixes for bunch of CVEs](https://go.dev/doc/devel/release#go1.20), Update go to build blackbox_exporter.

Also see #1070

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,44 +1,46 @@
 module github.com/prometheus/blackbox_exporter
 
-go 1.18
+go 1.20
```
Comment (SuperQ) [inline go.mod:3]: We only update the Go minimum version if required.

File: go.mod [hunk 2]
```diff
@@ -1,44 +1,46 @@
...
-	github.com/miekg/dns v1.1.50
+	github.com/miekg/dns v1.1.54
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.14.0
-	github.com/prometheus/client_model v0.3.0
-	github.com/prometheus/common v0.39.0
-	github.com/prometheus/exporter-toolkit v0.8.2
-	golang.org/x/net v0.7.0
-	google.golang.org/grpc v1.52.3
+	github.com/prometheus/client_golang v1.15.1
```
Comment (SuperQ) [inline go.mod:11]: We typically let dependabot take care of these things.

File: .circleci/config.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ executors:
   # Whenever the Go version is updated here, .promu.yml should also be updated.
   golang:
     docker:
-      - image: cimg/go:1.19
+      - image: cimg/go:1.19.9
```
Comment (SuperQ) [inline .circleci/config.yml:8]: We don't ever specify a specific version here, this tag always uses the latest point release. No need to change it.

**Review by SuperQ — CHANGES_REQUESTED** [review]
These changes don't actually do anything, as the final binary is produced with the Prometheus `golang-builder` docker image. This is controlled via promu.

**Comment by electron0zero** [comment]
closing this in favor of https://github.com/prometheus/blackbox_exporter/pull/1072

**Pull request closed without merging**